src/core/gui/PageView.cpp

```diff
@@ -853,36 +853,32 @@ void XojPageView::paintPageSync(cairo_t* cr, GdkRectangle* rect) {
         this->verticalSpace->paint(cr, rect, zoom);
     }
 
+
+    cairo_scale(cr, zoom, zoom);
     if (this->textEditor) {
         cairo_save(cr);
-        cairo_scale(cr, zoom, zoom);
         this->textEditor->paint(cr, rect, zoom);
         cairo_restore(cr);
     }
     if (this->selection) {
         cairo_save(cr);
-        cairo_scale(cr, zoom, zoom);
         this->selection->paint(cr, rect, zoom);
         cairo_restore(cr);
     }
 
     auto* pdfToolbox = this->xournal->getControl()->getWindow()->getPdfToolbox();
     if (auto* selection = pdfToolbox->getSelection(); selection) {
-        cairo_scale(cr, zoom, zoom);
         selection->paint(cr, rect, zoom, pdfToolbox->selectionStyle);
     }
 
     if (this->search) {
         cairo_save(cr);
-        cairo_scale(cr, zoom, zoom);
         this->search->paint(cr, rect, zoom, getSelectionColor());
         cairo_restore(cr);
     }
 
     if (this->inputHandler) {
         cairo_save(cr);
-        int dpiScaleFactor = xournal->getDpiScaleFactor();
-        cairo_scale(cr, 1.0 / dpiScaleFactor, 1.0 / dpiScaleFactor);
```

Comment:
Isn't scaling by `1.0 / dpiScaleFactor` different from scaling by `zoom`?

Reply:
Please add a doc comment somewhere that explicitly states this assumption ((maybe in PageView? not sure where an appropriate place would be). It will help greatly when we revisit the code!